### PR TITLE
Enhance CMS Validation Logic and Slug Handling, and Refactor Codebase for Maintainability

### PR DESCRIPTION
**Pull request by mirhamasala**
This Pull Request introduces a series of improvements and refactorings aimed at enhancing the robustness and flexibility of our content management system (CMS) and its validation logic. Below is a detailed summary of the changes:

### Key Changes

1. **Ensure the Right Identifier Fields**:
   - Adjusted the logic to ensure the correct identifier fields are used across the CMS.

2. **Remove Slug Fields from Content Entries**:
   - Removed explicit slug fields from content entries. These fields were in place as a validation check for when we migrated the data. It's no longer needed.
   - **Slug Handling**:
     - Removed manual slug fields and corresponding validation checks.

3. **Remove Slug Validation Check**:
   - Eliminated unnecessary validation checks for slug fields, simplifying the validation logic.
   - **Slug Handling**:
     - Removed manual slug fields and corresponding validation checks.

4. **Get Featured Entries with Utility Function**:
   - Using our utility function to fetch featured entries.
   - **Utility Functions**:
     - Added a utility function for fetching featured entries, improving code reuse.

5. **Add Slug to Markdown Data Based on Filename**:
   - Added logic to automatically generate slugs from filenames and include them to the data objects.
   - **Slug Handling**:
     - Automatic slug generation from filenames using `path.parse`.

6. **Refactor: Use path.parse to Extract Slug from Filename**:
   - Refactored the slug extraction logic to use `path.parse`.
   - **Slug Handling**:
     - Automatic slug generation from filenames using `path.parse`.

7. **Refactor: Simplify Category Data Extraction with Destructuring**:
   - Simplified the extraction of category data using destructuring.
   - **Code Simplification**:
     - Simplified category data extraction using destructuring.

8. **Remove Config Default**:
   - Removed unnecessary default configurations to clean up the codebase.
   - **Code Simplification**:
     - Cleaned up default configurations to reduce clutter.

9. **Determine Fields to Skip Check Dynamically Based on Required**:
   - Implemented dynamic determination of fields to skip during validation based on the `required` attribute.
   - **Validation Logic**:
     - Dynamic skipping of non-required fields during validation.

10. **Improve Front Matter Validation to Handle Nested Fields and Skip Non-required Fields**:
    - Enhanced the front matter validation logic to handle nested fields and properly skip fields where `required` is set to `false`.
    - **Validation Logic**:
      - Recursive validation for nested object fields.
      - Dynamic skipping of non-required fields during validation.

### CODE REVIEW

**Comment by barbaraperic** [comment]
Long description when creating ecosystem project

Screenshot 2024-07-10 at 11 17 41

**Comment by barbaraperic** [comment]
When creating an ecosystem project, I get an error on missing tags. Tho I don't see a tags field in the CMS 

Screenshot 2024-07-10 at 11 21 38

cms

**Comment by barbaraperic** [comment]
Creating **events** and **blogs** with only required fields works well 👌

File: src/app/_utils/validateFrontMatter.ts [hunk 1]
```diff
@@ -1,20 +1,51 @@
 import { CMSFieldConfig } from '@/types/cmsConfig'
 
+function validateField(
+  data: any,
+  field: CMSFieldConfig,
+  missingFieldsToSkipCheck: string[],
+  missingFields: string[],
+) {
+  if (field.widget === 'object' && field.fields) {
+    console.log('field.widget', field)
```
Comment (barbaraperic) [inline src/app/_utils/validateFrontMatter.ts:10]: console.log

**Comment by mirhamasala** [comment]
@barbaraperic 

>When creating an ecosystem project, I get an error on missing tags. Tho I don't see a tags field in the CMS

Thank you. Great catch! 🙏🏼 We are going to want to sunset the `tags` field because we are going to use `subcategories` in its place. You don't see the field in the CMS because these an option to hide it directly in the config - it's the `widget: hidden`

>Long description when creating ecosystem project

Known issue - We have a [ticket](https://www.notion.so/filecoin/FF-Ecosystem-Explorer-Projects-Include-max-characters-for-short-description-a57cef33acc54733aa73e55291fe514e?pvs=4) open!

**Pull request merged**